rust/src/file.rs

```diff
@@ -176,11 +188,24 @@ impl File {
             // fill / modify buffer
             buf_mat.copy_from(cur_dst_pos, src_mat, cur_src_box)?;
 
+            if self.header.block_type == BlockType::Raw {
```

Comment:
Hmm, not sure whether this case distinction is necessary. The `seek_block` method doesn't do any seeking if we're already at the right location. Ideally, we keep it simple and call `seek_block` no matter what.

Reply:
Good point! Nevertheless, we have to set `self.block_idx` conditionally here, since the seek method would otherwise try to calculate the correct position via the jump table which is not complete yet. I left the `self.block_idx = ...` line in and moved the seek method out of the if, since it will be a no-op if block_idx is already correct.